source/Sylvan.Data.Csv.Tests/CsvDataReaderTests.cs

```diff
@@ -2059,6 +2059,75 @@ public void LaxQuotesMid(string data, string expected)
 		Assert.Equal(expected, value);
 	}
 
+	[Fact]
+	public void ExceptionInit()
+	{
+		// When the malformed field is on the first row of data
+		// a different code path is used, so we need to test that independently.
+		var data =
+			"""
+			a,b,c
+			4,5,"6
+			""";
+
+		using var csv = CsvDataReader.Create(new StringReader(data));
+
+		var ex = Assert.ThrowsAny<CsvFormatException>(() => csv.Read());
+		Assert.Equal(1, csv.RowNumber);
+
+		// can read the two fields before the corrupt field
+		Assert.Equal("4", csv.GetString(0));
+		Assert.Equal("5", csv.GetString(1));
+		// but can't read the corrupt one or beyond.
+		Assert.ThrowsAny<InvalidOperationException>(() => csv.GetString(2));
+		Assert.ThrowsAny<InvalidOperationException>(() => csv.GetString(3));
+
+#if NET6_0_OR_GREATER
+		// Can read the raw record up to the beginning of the faulty field.
+		var s = csv.GetRawRecordSpan();
+		Assert.Equal("4,5,", s);
+#endif
+
+		// shouldn't be able to resume reading after an exception
+		Assert.ThrowsAny<InvalidOperationException>(() => csv.Read());
+		Assert.Equal(1, ex.RowNumber);
+	}
+
+	[Fact]
+	public void Exception()
```

Comment:
@loop-evgeny Take a look at this test and let me know if you think it covers the issue you reported in #262

Reply:
Yes, I think it does - thanks!